tests/cases/compiler/shadowedReservedCompilerDeclarationsWithNoEmit.ts

```diff
@@ -0,0 +1,53 @@
+// @target: es5
+// @noemit: true
+
+// Shadowed captured this and super
+class Base { }
+class C extends Base {
+    constructor() {
+        super();
+    }
+
+    foo() {
+        let _this = this;
+
+        () => {
+            _this;
+        };
+    }
+
+    bar() {
+        let _super = this;
+    }
+}
+
+
+/// shadowed arguments
+function f1(arguments, ...a) {
+}
+
+class C2 extends Base {
+    constructor() {
+        super();
+        var _newTarget = "";
+        var t = new.target;
+        var y = _newTarget;
+    }
+}
+
+
+// Shadowed Promise
+var Promise = null;
+async function f4() {
+    return 0;
+}
+
+
+// shadowed require
+var require = 0;

```

Comment:
are these actually shadowed? I thought they came from `@types/node`. In tests I've written, I've had trouble getting them to be *declared* instead of the other way round.

Reply:
it is just that the emitter expects them to exist and to mean something specific when `--module` is `AMD` or `CommonJS`. this ensures that users do not accidentally define a local variable with the same name.